backend/internal/models/admin.go

```diff
@@ -0,0 +1,51 @@
+package models
+
+// Struct to represent an admin
+type Admin struct {
+	ID             string `json:"id"` // the ID of the corresponding Firebase Auth user, not a UUID
+	Name           string `json:"name"`
+	Email          string `json:"email"`
+	CanManageUsers bool   `json:"can_manage_users"`
+	CanUploadBooks bool   `json:"can_upload_books"`
+	CanEditBooks   bool   `json:"can_edit_books"`
+	CanDeleteBooks bool   `json:"can_delete_books"`
+	IsPrimaryAdmin bool
+}
+
+// Struct to read in fields used during admin creation
+type CreateAdmin struct {
+	Name           string `json:"name"`
+	Email          string `json:"email"`
+	Password       string `json:"password"`
+	CanManageUsers bool   `json:"can_manage_users"`
+	CanUploadBooks bool   `json:"can_upload_books"`
+	CanEditBooks   bool   `json:"can_edit_books"`
+	CanDeleteBooks bool   `json:"can_delete_books"`
+	IsPrimaryAdmin bool
+}
```

Comment:
I still think there's some confusion here on `IsPrimaryAdmin`. To clarify:

- We do want to show if an admin is primary in `GET /admins`, so IsPrimaryAdmin should have a json: tag in Admin
- We do _not_ want to allow the is_primary_admin field in PATCH requests, so the IsPrimaryAdmin field should be deleted entirely from CreateAdmin. AFAIK this struct is not being used for anything other than parsing API requests, and since IsPrimaryAdmin should never be on an API request, it shouldn't be on the CreateAdmin struct.

Also, deleting the json: tag doesn't stop a field from being marshalled/unmarshalled to/from JSON - it will just use the name of the field as the key. We only specify the json: tags to change format - from CanManageUsers to can_manage_users for example.